unit_tests/sources/streams/http/test_http_client.py

```diff
@@ -744,3 +744,56 @@ def test_given_different_headers_then_response_is_not_cached(requests_mock):
     )
 
     assert second_response.json()["test"] == "second response"
+
+
+@pytest.mark.usefixtures("mock_sleep")
+@pytest.mark.parametrize(
+    "response_code, expected_failure_type, error_message, exception_class",
+    [
+        (400, FailureType.system_error, "test error message", UserDefinedBackoffException),
+        (401, FailureType.config_error, "test error message", UserDefinedBackoffException),
+        (403, FailureType.transient_error, "test error message", UserDefinedBackoffException),
+        (400, FailureType.system_error, "test error message", DefaultBackoffException),
+        (401, FailureType.config_error, "test error message", DefaultBackoffException),
+        (403, FailureType.transient_error, "test error message", DefaultBackoffException),
+        (400, FailureType.system_error, "test error message", RateLimitBackoffException),
+        (401, FailureType.config_error, "test error message", RateLimitBackoffException),
+        (403, FailureType.transient_error, "test error message", RateLimitBackoffException),
+    ],
+)
+def test_send_with_retry_raises_airbyte_traced_exception_with_failure_type(
```

Comment:
I like the fact that you added test coverage but I fear this test might be fragile i.e. that a change on how HttpClient is implemented might break this test even if this behavior does not change. This is caused by this test calling/mocking internal methods like `_send` and `_send_with_retry`.  Would there be a way to refactor this to ensure it is not as fragile? Can we use our HttpMocker to simulate the use cases?